src/pages/Onboarding/index.tsx

```diff
@@ -0,0 +1,280 @@
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Input } from "@/components/ui/input";
+import { TextareaAutosize } from "@mui/material";
+import { Button } from "@/components/ui/button";
+import { useForm } from "react-hook-form";
+import api from "@/api/baseApi";
+import { Loader2, CheckCircle2, XCircle } from "lucide-react";
+import { useSnackbar } from "@/contexts/SnackbarProvider";
+
+// SHADCN Dialog helpers
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogFooter,
+  DialogTrigger,
+  DialogClose,
+} from "@/components/ui/dialog";
+
+interface ProfileForm {
+  full_name: string;
+  username: string;
+  bio?: string;
+  birthday?: string;
+}
+
+const OnboardingProfile: React.FC = () => {
+  const navigate = useNavigate();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+    reset,
+    watch,
+  } = useForm<ProfileForm>({
+    defaultValues: {
+      full_name: "",
+      username: "",
+      bio: "",
+      birthday: "",
+    },
+  });
+
+  /* dialog state */
+  const [open, setOpen] = useState(true); // Set dialog to open by default
+  const timerRef = React.useRef<NodeJS.Timeout>();
+  const [popMessage, setPopMessage] = useState<{
+    ok: boolean;
+    text: string;
+  } | null>(null);
+
+  const showDialog = (ok: boolean, text: string) => {
+    setPopMessage({ ok, text });
+    setOpen(true);
+  };
+
+  // Age validation function (user must be at least 13 years old)
+  const isAbove13 = (birthday: string) => {
+    const birthDate = new Date(birthday);
+    const age = new Date().getFullYear() - birthDate.getFullYear();
+    return age >= 13;
+  };
```

Comment:
Birthday: 05/17/2012 -> This is 13 years - 1 day -> it should throw error 
This is due to this function not taking into account actual month and day